libnmstate/nm/connection.py

```diff
@@ -211,22 +229,13 @@ def _active_connection_callback(self, src_object, result, user_data):
         else:
             devname = nm_act_con.props.connection.get_interface_name()
             logging.debug(
-                'Connection activation initiated: dev=%s, con-state=%s',
+                'Connection activation initiated: dev=%s, con-uuid=%s, '
+                'con-state=%s',
                 devname,
+                nm_act_con.get_uuid(),
                 nm_act_con.props.state,
             )
-
-            ac = ActiveConnection(nm_act_con)
-            if ac.is_active:
-                self._mainloop.execute_next_action()
-            elif ac.is_activating:
-                self.waitfor_active_connection_async(ac)
-            else:
-                self._mainloop.quit(
-                    'Connection activation failed on {}: reason={}'.format(
-                        ac.devname, ac.reason
-                    )
-                )
+            self.wait_active_async()
```

Comment:
Just checking if I understand the reasoning here: We assume the state could have not changed already because we have not let the mainloop run yet and process the events (and update the state of the objects).

Reply:
The `wait_active_async()` return to mainloop after registered the `_active_timeout_callbak()`, so the mainloop is keep running.